### PR TITLE
Added drawer in swap for settings

### PR DESCRIPTION
**Pull request by prodesert22**
- Added input for slippage
- Added input for transaction deadline
- Added switch button for expert mode
Peek 07-04-2022 13-29

Added new component **NumberOptions**
image

### CODE REVIEW

**Comment by HariSeldon23** [comment]
Task linked: [CU-1xwdvrc No settings mode](https://app.clickup.com/t/1xwdvrc)

**Comment by HariSeldon23** [comment]
This looks awesome mate. I'll wait for Sarju to review it before we then merge into Dev branch

**Comment by HariSeldon23** [comment]
@StarlordPangolin and @ErichPNG this will hopefully go into Beta pretty soon and then be deployed to Production quite soon

**Comment by StarlordPangolin** [comment]
> 


> @StarlordPangolin and @ErichPNG this will hopefully go into Beta pretty soon and then be deployed to Production quite soon

Noted!

**Comment by ErichPNG** [comment]
Awesome. Will be great for a post

**Comment by SarjuHansaliya** [comment]
I am reviewing this PR today and if all looks good then will be merged and released new version

File: src/components/SwapWidget/LimitOrder/index.tsx
Comment (SarjuHansaliya) [inline src/components/SwapWidget/LimitOrder/index.tsx:536]: Now as I think, limit order should not have settings because its ultimately different thing as we are using galeto library. let me know if I am thinking differently.

File: src/components/NumberOptions/index.tsx
Comment (SarjuHansaliya) [inline src/components/NumberOptions/index.tsx:5]: add storybook for this component

File: src/components/NumberOptions/index.tsx
Comment (SarjuHansaliya) [inline src/components/NumberOptions/index.tsx:16]: remove Percent component from interface and use NumberOptions everywhere but make sure its working with new component because there is little difference in Percent and NumberOptions component

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]

File: src/components/SwapWidget/LimitOrder/index.tsx
Comment (prodesert22) [reply src/components/SwapWidget/LimitOrder/index.tsx]: Removed

File: src/components/NumberOptions/index.tsx
Comment (prodesert22) [reply src/components/NumberOptions/index.tsx]: Added

File: src/components/NumberOptions/index.tsx
Comment (prodesert22) [reply src/components/NumberOptions/index.tsx]: For me to remove do I need this component to be in this repo?

File: src/components/SwapWidget/Settings/index.tsx
Comment (prodesert22) [reply src/components/SwapWidget/Settings/index.tsx:103]: similar to this?
Peek 15-04-2022 08-01

Comment (SarjuHansaliya) [reply src/components/SwapWidget/Settings/index.tsx:103]: yes correct @prodesert22 and also functionality of export mode should work

File: src/components/NumberOptions/index.tsx
Comment (SarjuHansaliya) [reply src/components/NumberOptions/index.tsx]: NumberOptions is taken from Percent component. so ultimately we should be able to remove Percent component from interface because NumberOptions is almost same as Percent

Comment (prodesert22) [reply src/components/NumberOptions/index.tsx]: I created this PR for this https://github.com/pangolindex/interface/pull/447

File: src/components/SwapWidget/MarketOrder/index.tsx
Comment (SarjuHansaliya) [inline src/components/SwapWidget/MarketOrder/index.tsx:525]: When expert mode is enable at that time when user click on "Swap" button then spinner should come same as what we have currently. please see below screenshot

Screenshot 2022-04-20 at 5 32 14 PM

and once user confirm transaction in metamask then confirmation message should appear. please see below screenshot
Screenshot 2022-04-20 at 17-34-38 Pangolin

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]

**Pull request merged**
